proto/storage/v1/storage.proto

```diff
@@ -11,49 +11,75 @@ option csharp_namespace = "Nitric.Proto.Storage.v1";
 
 // Services for storage and retrieval of files in the form of byte arrays, such as text and binary files.
 service StorageService {
-  // Retrieve an item from a bucket
-  rpc Read (StorageReadRequest) returns (StorageReadResponse);
-  // Store an item to a bucket
-  rpc Write (StorageWriteRequest) returns (StorageWriteResponse);
-  // Delete an item from a bucket
+  // Retrieve a file from a bucket
+  rpc Read (StorageReadRequest) returns (stream StorageReadResponse);
+  // Store a file to a bucket
+  rpc Write (stream StorageWriteRequest) returns (StorageWriteResponse);
+  // Delete a file from a bucket
   rpc Delete (StorageDeleteRequest) returns (StorageDeleteResponse);
 }
 
-// Request to put (create/update) a storage item
-message StorageWriteRequest {
-  // Nitric name of the bucket to store in
+// Message containing data being written or read, with associated metadata.
+message StorageDataChunk {
+  // The contents of this chunk of data
+  bytes data = 1;
+  // TODO: confirm need for checksum data validation.
+  // CRC32C digest of this chunk.
+  // int32 crc32c = 2;
+}
+
+// Message containing all metadata for a file, including it's key, bucket, etc.
+message StorageFile {
+  // Nitric name of the bucket storing the file.
   //  this will be automatically resolved to the provider specific bucket identifier.
   string bucket_name = 1;
   // Key to store the item under
   string key = 2;
-  // bytes array to store
-  bytes body = 3;
 }
```

Comment:
Wondering whether StorageItem would be a better name for this message type. Having the item name (filename ?) and and content type, and length files would be extremely useful.  Note we can populate could of these values on the server, if not supported natively by the backing provider service.

Supporting a bucket list operation in the future would be extremely handy, and being able to fetch this meta data without the content.  People could then fetch the item content they are interested in.